src/psyclone/psyir/transformations/omp_loop_trans.py

```diff
@@ -131,6 +138,59 @@ def __init__(self, omp_directive="do", omp_schedule="auto"):
     def __str__(self):
         return "Adds an OpenMP directive to parallelise the target loop"
 
+    def _add_asynchronicity(self, node: Loop, instance: Directive):
+        '''
+        TODO
+        '''
+        # Only OMPDoDirective supports nowait of these directives. Needs to be
+        # exact type check
+        if type(instance) is not OMPDoDirective:
```

Comment:
Could this method belong to OMPDoDirective instead? (With the other directives just having an empty implementation?) That might be nonsense but this feels a bit weird.

Reply:
It would be difficult as its using parts of inherited transformations (that make more sense in other transformations) - its more just a special case of this Transformation doing a lot of different things I think (with different rules on the possible Directives it can create).

Reply:
Done